aten/src/ATen/native/Gesv.cpp

```diff
@@ -72,6 +78,63 @@ static void applyGesv(Tensor& b, Tensor& A, std::vector<int64_t> infos) {
   }
 }
 
+std::tuple<Tensor&,Tensor&> _gesv_single_out_cpu(
+    Tensor& sol, Tensor& lu,
+    const Tensor& self, const Tensor& A) {
+#ifndef USE_LAPACK
+  AT_ERROR("gesv: LAPACK library not found in compilation");
+#endif
+  int64_t bx = self.size(0);
+  int64_t by = (self.dim() == 1) ? 1 : self.size(1);
+  int64_t ax = A.size(0);
+  int64_t ay = A.size(1);
+  int info = 0;
+
+  /* Init to column major format. `sol` and `lu` need to be contiguous
+   * since we pass sol.data() and lu.data() to Lapack */
+  bool tc = sol.dim() == 2 && isTransposeContiguous(sol);
+  if (tc) {
+    // if transpose of output tensor is contiguous, use it
+    sol.t_();
```

Comment:
I'm not understanding why this matters? Can you enlighten me? It looks like `sol` can be `resize_`d below anyways, right?

Reply:
<img width="511" alt="screen shot 2018-08-01 at 4 46 01 pm" src="https://user-images.githubusercontent.com/1777276/43554652-722cc238-95aa-11e8-93f7-01bd731a227c.png">
So in the test cases and the documentation the Tensors are initialized and then transposed before being passed to `torch.gesv`. We have to transpose it internally as well for Lapack/Magma's column major format and we need the data at`A_ptr` and `b_ptr` to be contiguous before we pass it. If a user calls `torch.gesv(b, A, out=(b, A))` and has transposed the input Tensor since creating it, we can just transpose `sol` here and use it directly.  The resize will not have any effect if it is already the required size (eg. the above call) but it will take care of Tensors whose transpose is contiguous but size is incorrect. Also, this helps with reusing output tensors;

for example:

    ta = torch.Tensor()
    tb = torch.Tensor()
    torch.gesv(b, a, out=(ta, tb)) # this will work
    torch.gesv(b, a, out=(ta, tb)) # this will not
The second time `gesv` is called, ta and tb will have been generated as a result of Lapack/Magma and transposed before being returned (our convention is to always return the transpose according to documentation). So even though they may have the correct size (resize_ won't work as expected on a non-contiguous Tensor), we should use their transpose if it is contiguous since we need pointers to contiguous data.

Reply:
Thanks. But doesn't this still break if `sol` is not transpose contiguous but have size `{bx, by}`? I don't see any code handling the logic that in that case (1) applying lapack gesv on a buffer and then (2) copy to `sol`. Am I missing anything?